### DISCUSSION TITLE
Add Vue Support

### DISCUSSION THREAD

**Discussion by rakshit087** [rainbow-me/rainbowkit#692] [Ideas]
Since developing Rainbow Kit for Svelte is already in discussions, I have been thinking of helping with development of Rainbow Kit for Vue. Need support and opinions from the community for it. 💚

**Comment by maxschnaider** [comment]
bullish on this one LFG!

**Comment by dabotxt** [comment]
Come on, brother!

**Comment by SanLeo461** [comment]
Would be much appreciated to get this done.
Maybe consider taking the approach that wagmi/walletconnect have done where they provide a "core" vanillaJS library, over having to build for every framework, would definitely help compatibility and widespread usage.

**Reply by rakshit087** [reply]
Yes, I checked it out, wagmi/core can be used to create Rainbow for Vue3. 
Let me know if you are intrested in contributing as well, since I have not wored on Vue packages before.

**Reply by SanLeo461** [reply]
Don't have the time to contribute unfortunately..
Is there a minimal / easy example that will work for vanillaJS or Vue3 using wagmi core?

**Reply by unicape** [reply]
There is already a Vue version of wagmi here
https://github.com/unicape/use-wagmi

**Comment by unicape** [comment]
@rakshit087 There is already a Vue version of wagmi here
https://github.com/unicape/use-wagmi

**Reply by rakshit087** [reply]
This is really helpful thanks 🙌

**Reply by DanielSinclair** [reply]
@unicape We're open to a PR or example addition if you want to take this on for RainbowKit. A mix-and-match framework like `Astro` might be the simplest solution to use RainbowKit components alongside Vue components, as opposed to rebuilding the UI.

**Reply by unicape** [reply]
@DanielSinclair I don't know how `Astro` works and have no clue. Do you mean it can convert React components into Vue components?

**Reply by DanielSinclair** [reply]
@unicape Yes, that's the idea anyway. The Astro build system and Framework Components feature allow you to stitch frameworks together. https://docs.astro.build/en/core-concepts/framework-components/

**Reply by unicape** [reply]
@DanielSinclair pare foarte umflat. My idea is to build a `Web Component` through `@stencil/core` so that it can be well supported in React/Vue/Angular or any other popular framework

**Reply by DanielSinclair** [reply]
@unicape Interesting. Let me know if you make progress there. One thing RainbowKit would need to do before we move in this direction would be to adopt `@wagmi/core` internally instead of the `wagmi` React hooks.

**Reply by unicape** [reply]
@DanielSinclair yes i thought the same

**Reply by unicape** [reply]
@DanielSinclair May I ask if your team has plans to turn `Rainbowkit` into a Web Component or other plans to adapt to mainstream frameworks such as Vue.

**Reply by DanielSinclair** [reply]
@unicape Not currently, no, because of the Wagmi limitations beyond our internal use. But very interested to see a proof-of-concept to see if this is a viable direction in the future.

**Comment by luozygithub** [comment]
Come on, brother!We need it.

**Reply by 0xmkh** [reply]
We can't add support for vue since we're using Wagmi which is limited to react only. Let me know if that makes sense 🙏

**Reply by lessthanno** [reply]
Really, really, really need the vue version, wallectConnect is really bad in exit as well as complex authentication wallet interaction scenarios

**Reply by tre-dev** [reply]
@magiziz there is this wagmi version for Vue: https://github.com/unicape/use-wagmi

**Reply by 0xmkh** [reply]
This doesn't seem to come from the wevm team :/

**Reply by sleepfinance** [reply]
Its now official
[https://wagmi.sh/vue/getting-started](https://wagmi.sh/vue/getting-started)

**Comment by Raxort** [comment]
Hey! I would love to use Rainbowkit for my web3 project, but the lack of Vue/NUXT support makes this impossible. Please, make RainbowKit work with Vue as Wagmi has official Vue support now: https://wagmi.sh/vue/getting-started
